bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -594,18 +991,56 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
             var me = this;
             var filterFunction = me.layerFilters[filterId];
             var allLayers = me.getAllLayers();
-            if(!filterFunction) {
+            if (!filterFunction) {
                 Oskari.log(this.getName()).warn('[MapLayerService] not found layer filter "' + filterId + '". Returning all layers.');
                 return allLayers;
             }
             var filteredLayers = [];
-            allLayers.forEach(function(layer){
-                if(filterFunction(layer)) {
+            allLayers.forEach(function(layer) {
+                if (filterFunction(layer)) {
                     filteredLayers.push(layer);
                 }
             });
             return filteredLayers;
         },
+        /**
+         * @method  @public getFilteredLayers  Get filtered layer groups
+         * @param  {String} filterId filter id
+         * @return {Array}   filtered layers list, if not found filter by id then return all layers
+         */
+        getFilteredLayerGroups: function(filterId) {
+            var me = this;
+            var filterFunction = me.layerFilters[filterId];
+            var allLayerGroups = me.getAllLayerGroups();
+
+            if (!filterFunction) {
+                Oskari.log(this.getName()).warn('[MapLayerService] not found layer filter "' + filterId + '". Returning all layer groups.');
+                return allLayerGroups;
+            }
+            var filteredLayerGroups = [];
+            allLayerGroups.forEach(function(group) {
+                var filteredLayers = [];
+                group.layers.forEach(function(layer) {
+                    var mapLayer = me.createMapLayer(layer);
```

Comment:
The name suggests some kind of filtering, but layers are being created here. What's going on?